packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -403,13 +403,51 @@ gmaps.PolylineOptions _polylineOptionsFromPolyline(
   final List<gmaps.LatLng> paths =
       polyline.points.map(_latLngToGmLatLng).toList();
 
+  final bool hasPattern = polyline.patterns.isNotEmpty;
+
+  final List<gmaps.IconSequence> iconSequences = <gmaps.IconSequence>[];
+
+  if (hasPattern) {
+    List<Object>? getPattern(List<List<Object>> input, String patternName) {
+      final List<Object> found = input.firstWhere(
+        (List<Object> innerList) =>
+            innerList.isNotEmpty &&
+            innerList[0] is String &&
+            innerList[0] == patternName,
+        orElse: () => <Object>[],
+      );
+      return found.isEmpty ? null : found;
+    }
+
+    final List<List<Object>> patternJson = polyline.patterns
+        .map((PatternItem element) => element.toJson() as List<Object>)
+        .toList();
+
+    final List<Object>? dotPattern = getPattern(patternJson, 'dot');
+    final List<Object>? gapPattern = getPattern(patternJson, 'gap');
+
+    final gmaps.GSymbol icon = gmaps.GSymbol()
+      ..path = dotPattern != null ? gmaps.SymbolPath.CIRCLE : 'M 0,-1 0,1'
+      ..strokeOpacity = _getCssOpacity(polyline.color)
+      ..strokeColor = _getCssColor(polyline.color)
+      ..scale = 4;
```

Comment:
This is discarding the desired length of the line in case of a dash, and the width of the circle, which should be the weight of the polyline (and not be hard-coded to `4`).